index.js

```diff
@@ -804,37 +846,20 @@ class Command extends EventEmitter {
   _parseExpectedArgs(args) {
     if (!args.length) return;
     args.forEach((arg) => {
-      const argDetails = {
-        required: false,
-        name: '',
-        variadic: false
-      };
-
-      switch (arg[0]) {
-        case '<':
-          argDetails.required = true;
-          argDetails.name = arg.slice(1, -1);
-          break;
-        case '[':
-          argDetails.name = arg.slice(1, -1);
-          break;
-      }
-
-      if (argDetails.name.length > 3 && argDetails.name.slice(-3) === '...') {
-        argDetails.variadic = true;
-        argDetails.name = argDetails.name.slice(0, -3);
-      }
-      if (argDetails.name) {
-        this._args.push(argDetails);
-      }
+      const argDetails = parseArg(arg);
+      argDetails && this._args.push(argDetails);
```

Comment:
Still adding argDetails to `_args` here.

Reply:
can you explain me what is the case were argDetails is undefined? I see it's something with the arg name, in which case an argument cannot have a name?

Reply:
In the original `_parseExpectedArgs`, if an argument is not in the expected format then the argument is silently ignored because the name does not get set. For example: `{oops}` or `foo` or `<...>`. 

In the refactored code that leads to undefined being returned from `parseArg`.

Reply:
So I wonder what we should do in that case if we use the Argument class...
It's a bit weird that the initialization of the class fails silently. I would say that it's better to throw an error if the argument is invalid, but it's a breaking change, so i think we should do it later and I'll try to work around it for now.

Reply:
Ok, after thinking about it again I realized that it's not a breaking change because we can do it only for the new synax. So what do you say? Do you think we should throw on bad argument?

Reply:
Yes. Throw on bad programmer input is a direction we are slowly moving.

Reply:
@shadowspawn I have added a throw inside the class, but I can't solve this comment because we don't want to introduce a breaking change so I still need to use argDetails instead of the Argument class here.
when we deprecate `arguments` prop we could remove all usage of argDetails.